lib/kcerts/cache.go

```diff
@@ -0,0 +1,63 @@
+package kcerts
+
+import (
+	"errors"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/cache"
+	"github.com/enfabrica/enkit/lib/config/marshal"
+	"path/filepath"
+)
+
+const (
+	SSHCacheKey  = "enkit_ssh_cache_key"
+	SSHCacheFile = "ssh.json"
+)
+
+var (
+	SSHAgentNoCache      = errors.New("the cache had not existed before")
+	SSHAgentCacheInvalid = errors.New("the cache is corrupted or invalid json")
+)
+
+func FetchSSHAgentFromCache(store cache.Store) (*SSHAgent, error) {
+	sshEnkitCache, isFresh, err := store.Get(SSHCacheKey)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching cache: %w", err)
+	}
+	defer store.Rollback(sshEnkitCache)
+	agent := &SSHAgent{}
```

Comment:
Following my suggestion below about a "Close" method:

`&SSHAgent{Close: func() {}}`

(when the agent is read from cache, there's nothing to close)